lib/utils/input-source-map-tracker.js

```diff
@@ -63,15 +66,26 @@ function fromSource(self, data, whenDone, context) {
       context.files.pop();
     } else if (nextAt == mapMatch.index) {
       var isRemote = /^https?:\/\//.test(sourceMapFile) || /^\/\//.test(sourceMapFile);
+      var isDataUri = DATA_URI.test(sourceMapFile);
       if (isRemote) {
         return fetchMapFile(self, sourceMapFile, context, proceedToNext);
       } else {
         var sourceFile = context.files[context.files.length - 1];
-        var sourceDir = sourceFile ? path.dirname(sourceFile) : self.options.relativeTo;
-        var sourceMapPath = path.resolve(self.options.root, path.join(sourceDir || '', sourceMapFile));
-
-        var sourceMapData = fs.readFileSync(sourceMapPath, 'utf-8');
-        self.trackLoaded(sourceFile || undefined, sourceMapPath, sourceMapData);
+        var sourceMapPath, sourceMapData;
+        if(isDataUri) {
+          // source map's path is the same as the source file it comes from
+          sourceMapPath = path.resolve(self.options.root, sourceFile);
+          var buffer = dataUriToBuffer(sourceMapFile);
```

Comment:
It's your decision, depending on your philosophy about dependencies.  Given
that they're small, simple libraries, I don't see a practical downside to
using them.

You'll also need to handle data URIs that are not base64 encoded.

On Sun, Jun 14, 2015 at 12:37 PM, Jakub Pawlowicz <notifications@github.com>
wrote:

> In lib/utils/input-source-map-tracker.js
> <https://github.com/jakubpawlowicz/clean-css/pull/600#discussion_r32381526>
> :
>
> >        if (isRemote) {
> >          return fetchMapFile(self, sourceMapFile, context, proceedToNext);
> >        } else {
> >          var sourceFile = context.files[context.files.length - 1];
> > -        var sourceDir = sourceFile ? path.dirname(sourceFile) : self.options.relativeTo;
> > -        var sourceMapPath = path.resolve(self.options.root, path.join(sourceDir || '', sourceMapFile));
> > -
> > -        var sourceMapData = fs.readFileSync(sourceMapPath, 'utf-8');
> > -        self.trackLoaded(sourceFile || undefined, sourceMapPath, sourceMapData);
> > +        var sourceMapPath, sourceMapData;
> > +        if(isDataUri) {
> > +          // source map's path is the same as the source file it comes from
> > +          sourceMapPath = path.resolve(self.options.root, sourceFile);
> > +          var buffer = dataUriToBuffer(sourceMapFile);
>
> I'm pretty sure we can do without those 2 dependencies given what they do:
> https://github.com/TooTallNate/node-data-uri-to-buffer/blob/master/index.js#L44
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jakubpawlowicz/clean-css/pull/600/files#r32381526>.
>